### ISSUE TITLE
Can not rename files of the torrent payload

### ISSUE THREAD

**Issue by slrslr** [Tribler/tribler#3455]
##### Tribler version/branch+revision: #####
7.0.0 x64 downloaded today from tribler.org, previously i used alpha version so i have %appdata%/.Tribler from previous version. Before alpha i had 6.x version.
##### Operating system and version: #####
Windows 10 64bit
##### Issue: #####
Hello, when i go to Downloads section, i see my torrents there and i try to click one, then on the Files tab i see wrong name of the file. On HDD i have different name and i want to keep the name as is on HDD, So i need to rename the file in Tribler. How can i do it please? If not possible, i think this can be good feature. If i rename file in Tribler, it should rename file on HDD and if on HDD is already renamed, then it will just update name in Tribler. Thank you

**Comment by devos50** [comment]
This functionality is not supported in Tribler 7 so far. We will consider adding it. Thank you for your suggestion!

**Comment by Dmole** [comment]
It's better to leave that sort of feature to the file system;
`mklink /H old.file  new.file`
or on UNIX like systems
`ln old.file  new.file`
https://en.wikipedia.org/wiki/Hard_link

**Comment by slrslr** [comment]
@Dmole It may be better from technical side of view, but certainly NOT better from the user point of view. User works via GUI, he is not a technical head who opens a terminal and spend time composing exact command for each file rename.

**Comment by Dmole** [comment]
@slrslr lol you typed 44 words to say that you can't type 3?

Anyway maybe this can tie into the new move feature
https://github.com/Tribler/tribler/issues/2901#issuecomment-493388260

**Comment by ichorid** [comment]
Unfortunately, it is not really possible to rename files in a torrent because it will technically become a new torrent, and will lose the connection to the swarm. On the other hand, the proper way to solve improperly shown names is to show them properly. Renaming in-GUI only does not solve anything.
So, closing this one. UTF-8 and Python3 support should fix all inconsistent names problems for us.

**Comment by slrslr** [comment]
> really possible to rename files in a torrent because it will technically become a new torrent
Unsure if it is mean not possible in Tribler or not possible in torrent client.
In case not possible at all, **you are wrong** - this feature is already present in modern clients like qBittorrent and uTorrent and more. This feature is important, because it does not lock users using stupid/incomplete file/folder names and allows user to seed multiple torrents having one common file/folder.

Either reopen this or allow new feature which will automaticaly (on background?) find already downloaded files on one's HDD and match it to the torrent despite of the different file/folder names. Please kindly see: https://github.com/qbittorrent/qBittorrent/issues/6520#issuecomment-458275109
@devos50

**Comment by ichorid** [comment]
@slrslr , you're right. In qBittorrent, it is possible to rename torrents and files in them. However, these changes are local to the user's PC and can never propagate to the network (because that essentially means seeding a new torrent). Also, if the user deletes the torrent (but keeps the files), if the torrent is downloaded again, the torrent client will not associate the previously renamed files with the files in the torrent. Thus, the file renaming feature is incompatible with incremental releases that are used, e.g. by some podcasts. The only reason for one to use this feature is to better organize the torrents' contents on disk (e.g. music collections).

Having said all that, there are functions in Libtorrent to do this: `torrent_handle::rename_file()` and `torrent_handle::remap_files()`. We'll try to implement the feature at some time. I am reopening the ticket.

**Comment by slrslr** [comment]
after 3 years.. v7.9.0 i see i am unable to rename files and folders when adding torrent

**Comment by ichorid** [comment]
@xoriole , would you consider taking this issue? :wink:

**Comment by slrslr** [comment]
Almost next 2 years passed.

**Comment by slrslr** [comment]
Another 2,5 years passed.. Ability to rename file/folder is very important.

**Comment by qstokkink** [comment]
I'll put this on the agenda for 8.3.0. We can expose [`rename_file()`](https://www.libtorrent.org/reference-Torrent_Info.html#rename-file).

On a technical level, the big caveat here is that if we accidentally supply an absolute path instead of a relative path, the file becomes detached from the torrent, which would cause `move_storage()` to no longer work. So, in the implementation of this we need to be very careful (or risk littering files all over users' machines).

**Comment by qstokkink** [comment]
I finally got around to testing `rename_file()`. It does not work as I expected.

Whereas the file is correctly renamed in the torrent information and shown in Tribler with that name, the actual downloaded files are not renamed (I renamed the bittorrent v2 test torrent's files to `othername*` here):

<img width="3063" height="1368" alt="Image" src="https://github.com/user-attachments/assets/4984a885-87b5-42a4-a905-dbc975e0e4ad" />

We'll need something else (either complementary or completely different).

---

`remap_files` also doesn't work for v2 torrents. It seems that there is no natural way forward (you run into what @ichorid mentioned with generating entirely new torrents - seeding to nobody). I'll put this on the back burner again, until someone has a good/reasonable idea on how to implement this.
